tests/unit/test_external_repo.py

```diff
@@ -77,3 +77,28 @@ def test_subrepo_is_constructed_properly(
             assert repo.config["remote"]["auto-generated-upstream"][
                 "url"
             ] == str(main_cache)
+
+
+@pytest.mark.parametrize("jobs", [1, 2, 4])
+def test_fetch_external_repo_jobs(
```

Comment:
Looks like the tests are failing. Also, these are pretty heavy for a unit tests. We could mock-out ExternalRepo and check that the things are passed. But if you don't feel like putting that much effort into it, we could merge without the tests too in this case. That part of external repo will be refactored in the near future.

Reply:
@efiop Thank you. 
Seems that it fails on some versions of python( 3.6, 3.7). The spy didn't get correct call arguments. 

And excuse me, I'm afraid that I didn't understand what does
`We could mock-out ExternalRepo and check that the things are passed.` mean.
Is there any example? Thank you.

Reply:
[`call_args.kwargs` is introduced after python3.8](https://bugs.python.org/issue21269)

Reply:
@karajan1001 Thank you for your patience. On second look, mocking out this might be a bit complex. Let's do another simpler test instead...